source/opt/fix_func_call_arguments.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2022 AMD LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _VAR_FUNC_CALL_PASS_H
+#define _VAR_FUNC_CALL_PASS_H
+
+#include "source/opt/pass.h"
+
+namespace spvtools {
+namespace opt {
+class FixFuncCallArgumentsPass : public Pass {
+ public:
+  FixFuncCallArgumentsPass() {}
+  const char* name() const override { return "fix-for-funcall-param"; }
+  Status Process() override;
+  // Query module whether function number is one
+  bool HasNoFunctionToCall();
+  // Replace function call argument of accesschain with memory variables
```

Comment:
In these function comments, you will want to say what each parameter is.  It is not clear how this comment fits with the comment.  There are also lots of implict connections between the parameters that would have to be called out.  For example, `operand` is the operand in `func_call_inst` at index `operand_index`.  Not clear if it is the in-operand index or the operand index.

If you simplifed the interface the way I suggested above a good comment might be:

> Copies from the memory pointed to by |operand_inst| to a new function scope variable before |func_call_inst|, and copies the value of the new variable back to the memory pointed to by |operand_inst| after |funct_call_inst|.  Returns the id of the new variable.